### DISCUSSION TITLE
Postfix as docker container

### DISCUSSION THREAD

**Discussion by platanax** [simple-login/app#443] [Self-hosting question]
Hi,
Any chance that postfix could be installed as a Docker container ? I tried it myself on my NAS resulting in an openrelay install...
Thanks !

**Comment by nguyenkims** [comment]
Yes! We are working on a better way to self host SimpleLogin via docker-compose where all components are run as Docker containers. This project can be tracked in this pull request https://github.com/simple-login/app/pull/355

This relies on a custom Postfix docker image that's also included in SimpleLogin GitHub https://github.com/simple-login/simplelogin-postfix-docker

**Reply by EricGuic** [reply]
Hi, here another french friend who have just discovered your great SimpleLogin !
First of all, thanks for this achievement, for an ever growing "self-hosted" world where privacy concerns gains more and more lights.

I'm also very interesting in installing SimpleLogin at home. But being not a developper, I need simple tutorial too. 
Here are an example of easy step-by-steps guide that a lot of Synology NAS owner are reading and following to install self-hosted solution (this example is about bitwarden, with Docker container).
https://mariushosting.com/how-to-install-bitwarden-on-your-synology-nas/
(and even better for newbies, the 30s install via a script : https://mariushosting.com/synology-30-second-dozzle-install-using-task-scheduler-docker/)

Are you aiming to provide an easy solution like this ?
Another solution could be through a manager like Portainer :
https://documentation.portainer.io/v2.0/stacks/create/

All of this is to allow more pepole to self-host SimpleLogin even if they don't know ssh command and other (too) complicated step-by-step from developpers.

Hope you agree with me. Best regards from Poitiers ;-)

**Reply by nguyenkims** [reply]
Salut Eric! I didn't know about Synology, it looks great for self-hosting software on home servers! 

SimpleLogin is a bit more complex than Bitwarden or Dozzle as it involves some DNS setup and requires several components (Postfix, Database, Web app, Email handler). Once Postfix can be run as a Docker container, we should be able to run SimpleLogin via Docker-compose and therefore on Portainer I'm not sure about whether Synology supports Docker compose though.

As we aren't expert on Portainer and Synology, we'd need a contributor who's familiar with these software to help transform the Docker-compose file to a format compatible with Portainer/Synology :).

**Reply by 0x5ECF4ULT** [reply]
Hi. Regarding this issue I've taken some time and tried to basically replicate your custom `postfix-pgsql` image. The only difference between your solution and mine is that I compile `postfix` from source. Take a look at it [here](https://github.com/0x5ECF4ULT/docker_postfix-pgsql)!

**Reply by 0x5ECF4ULT** [reply]
@EricGuic I accomplished something you may be interested in [here](https://gist.github.com/0x5ECF4ULT/5df0e29c99384c216b8f941bdd514b77).
